packages/modular-scripts/src/utils/getPackageDependencies.ts

```diff
@@ -69,26 +71,41 @@ export async function getPackageDependencies(
     targetManifest.dependencies,
   ) as Dependency;
 
-  const lockDeps = await buildDepTree(
-    // Build a dependency tree from the lockfile, using target dependencies and root dependencies in order of specificity
-    JSON.stringify(
-      Object.assign(Object.create(null), targetManifest, {
-        dependencies: Object.assign(
-          Object.create(null),
-          rootManifest.dependencies,
-          targetManifest.dependencies,
-        ) as Dependency,
-        devDependencies: Object.assign(
-          Object.create(null),
-          rootManifest.devDependencies,
-          targetManifest.devDependencies,
-        ) as Dependency,
-      }),
-    ),
-    lockFile,
-    true,
-    LockfileType.yarn,
-  );
+  let lockDeps: Dependency = {};
+
+  try {
+    // Try to parse with v1 parser
+    const parsedLockfile = lockfile.parse(lockFile);
+    lockDeps = Object.entries(deps).reduce<Record<string, string>>(
+      (acc, [name, version]) => {
+        acc[name] = (parsedLockfile.object as LockFileEntries)[
+          `${name}@${version}`
+        ].version;
+        return acc;
+      },
+      {},
+    );
+  } catch (e) {
+    // Try to parse as yaml (v2+) - https://github.com/yarnpkg/yarn/issues/5629#issuecomment-753418765
+    const parsedLockfile = yaml.load(lockFile) as LockFileEntries;
+    const dependencyArray = Object.entries(deps);
```

Comment:
What's the difference between `dependencyArray` and `Object.entries(parsedLockfile)`? I don't quite understand what this `O(n^2)` algo is doing.

Reply:
Why do we not need to add the dependencies to the right of the `parsedLockfile` entries after a match is found?

When you extract this to a separate function, I'd recommend adding a little context so it's easier for the next person who reads this to get to grips with the why. I'm not familiar with the new lockfile format!

Reply:
Refactored + added additional comments.